nipype/pipeline/engine/nodes.py

```diff
@@ -367,7 +368,22 @@ def run(self, updatehash=False):
             except:
                 os.remove(hashfile_unfinished)
                 raise
-            shutil.move(hashfile_unfinished, hashfile)
+
+            t = time()
+            timeout = float(self._config['execution']['job_finished_timeout'])
+            timed_out = True
+            while True:
+                try:
+                    shutil.move(hashfile_unfinished, hashfile)
+                    timed_out = False
+                    break
+                except Exception as e:
```

Comment:
Yes, otherwise we have a race condition between two processes/threads in a section where only one process is intended to be running. And I can't see how that could happen.